tools/cldr-apps/src/main/java/org/unicode/cldr/web/SurveyMain.java

```diff
@@ -2516,36 +2511,19 @@ public void close() {
             }
         }
 
-        public void internalClose() {
-            this.dbSource = null;
-        }
+        public void internalClose() {}
 
         public boolean isClosed() {
-            return this.dbSource == null;
+            return (use == 0);
         }
 
         public UserLocaleStuff(CLDRLocale locale) {
             synchronized (allUserLocaleStuffs) {
                 allUserLocaleStuffs.add(this);
             }
-
-            // TODO: refactor.
-            if (cldrfile == null) {
-                resolvedSource = getSTFactory().makeSource(locale.getBaseName(), true);
-                dbSource = resolvedSource.getUnresolving();
-                cldrfile =
-                        getSTFactory()
-                                .make(locale, true)
-                                .setSupplementalDirectory(getSupplementalDirectory());
-                resolvedFile = cldrfile;
-            }
         }
 
-        public void clear() {
-            // TODO: try just kicking these instead of clearing?
-            cldrfile = null;
-            dbSource = null;
-        }
+        public void clear() {}
```

Comment:
is `clear()` still called anywhere? could just remove it?

Reply:
I'm not sure, I didn't add it in this PR

Reply:
no callers, not even from .jsp

Reply:
I think UserLocaleStuff was for convenience of JSPs and other very old flows… to make it easier to get the 'current' CLDRFile.  Probably most of the class is obsolete.